scripts/globals/spells/trust/kupipi.lua

```diff
@@ -22,19 +26,28 @@ function onSpellCast(caster, target, spell)
 end
 
 function onMobSpawn(mob)
-    mob:addListener("COMBAT_TICK", "KUPIPI_COMBAT_TICK", function(trust, master, target)
-        function round(x)
-            return x >= 0 and math.floor(x + 0.5) or math.ceil(x - 0.5)
-        end
-        
-        local party = master:getPartyWithTrusts()
-        local lvl = trust:getMainLvl()
-        local maxCure = utils.clamp(round(lvl/8), 1, 6)
-
-        for _, member in ipairs(party) do
-            if member:getHPP() <= 75 then
-                trust:castSpell(maxCure, member)
-            end
-        end
-    end)
+    local CURE_I = 1
+    local FLASH  = 112
+    local ERASE  = 143
+    local DISPEL = 260
+
+    mob:addBehaviour(PARTY, HPP_LTE, 25, MA, SELECT_HIGHEST, tpz.magic.spellFamily.CURE)
+
+    mob:addBehaviour(PARTY, STATUS, tpz.effect.SLEEP_I, MA, SELECT_SPECIFIC, CURE_I)
+    mob:addBehaviour(PARTY, STATUS, tpz.effect.SLEEP_II, MA, SELECT_SPECIFIC, CURE_I)
+
+    mob:addBehaviour(PARTY, HPP_LTE, 75, MA, SELECT_HIGHEST, tpz.magic.spellFamily.CURE)
```

Comment:
why is there two behaviors for CURE? shouldnt the one for HPP_LTE, 75 be enough? (sorry for my blunt question, just trying to provide feedback :cat: )

Reply:
Useful question!

These behaviours are looked at in order behind the scenes, so if you wanted to weight a certain behaviour to be more important, you would put it higher on the list, or add another copy of it with different parameters. 

In this case, if someone in the party is critically low, Kupipi will try and drop a heal on them above all else. If not, she'll try to wake people up with Cure I, then she'll go to her regular type healing checks of 75% and all the other things in the list.

Reply:
Awesome that makes sense @zach2good  :smile_cat: :  can't wait to try it out :smile_cat: